Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48627 Refactored OpInfo testing to support custom SampleInputs, added addmm to op_db to test
* #47696 Support for OpInfo-based testing for operators in JIT
* #48567 Refactored assertAutodiff test to have better error message
* **#47695 Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic**

The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests

Differential Revision: [D25212437](https://our.internmc.facebook.com/intern/diff/D25212437)

Commit:
Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic

[ghstack-poisoned]

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests 

[ghstack-poisoned]

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests 

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/jit_metaprogramming_utils.py:
```diff
@@ -243,6 +243,8 @@ def get_call(method_name, func_type, args, kwargs):
         call = '{}.{}({})'.format(self_arg, method_name, argument_str)
     elif func_type == 'nn_functional':
         call = 'torch.nn.functional.{}({})'.format(method_name, argument_str)
+    elif func_type == 'function':
```
function vs functional?

mruberry commented:
This looks good to me, @eellison would you like to look at it from the JIT perspective?

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -0,0 +1,168 @@
+            or all([any(g.findNode(n) is not None for g in fusion_subgraphs) for n in fusible_nodes])
+
+        err_msg = "Failure in testing node's autodifferentiation, "
+        if should_autodiff_node:
```
We need to expand these error messages so they enumerate ALL the failures. That is, "found node X that should be fused but was not, found node Y that was fused that shouldn't be..." and suggest the appropriate remedy, "Add X to the list of fusible nodes..., Add Y to the list of nonfusible nodes..."

Lilyjjo replied:
Added specific error messages to the commit above this one in the stack

eellison commented on torch/testing/_internal/common_jit.py:
```diff
@@ -0,0 +1,168 @@
+            err_msg += "one or more nodes were expected to be autodiffed, " \
+                "but weren't found in specified fusible/nonfusible DifferentiableGraph group"
+        else: 
+            err_msg += "one or more nodes weren't expected to be autodiffed, " \
```
These error messages are much better! in the future, IMO it's nice to separate out mechanical movement of code from real changes

Lilyjjo replied:
Good suggestion, I added the error message changes to a diff on top of this one

eellison reviewed (commented):
lgtm - seems like it's mostly just code movement

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests 

[ghstack-poisoned]

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests 

[ghstack-poisoned]

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests

Differential Revision: [D25212437](https://our.internmc.facebook.com/intern/diff/D25212437)

[ghstack-poisoned]

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests

Differential Revision: [D25212437](https://our.internmc.facebook.com/intern/diff/D25212437)

[ghstack-poisoned]

mruberry reviewed (approved):
Looks good to me. ping @eellison since this is a refactor of jit test code.

Commit:
Update on "Refactor existing JIT testing utils to enable new OpInfo test suite to reuse existing logic"


The method_tests from common_methods_invoations.py are being migrated into a new OpInfo class-based testing framework. The work in this commit pulls out the functions embedded in the old method_tests logic and places them in a location that both the old method_tests and OpInfo tests can use

Specifically: created torch/testing/_internal/common_jit.py from functions and methods in torch/testing/_internal/jit_utils.py and test/test_jit.py. Also created new intermediate class JitCommonTestCase to house moved methods. Also slightly modified jit_metaprogramming_utils.py to work for OpInfo tests

Differential Revision: [D25212437](https://our.internmc.facebook.com/intern/diff/D25212437)

[ghstack-poisoned]
